refactor(ast): `StaticMemberExpression::get_first_object` use loop instead of recursion

overlookmotel opened the pull request:
Follow-on after #6969. Pure refactor. Loops are generally cheaper than recursion.

Dunqing reviewed (approved):
I am curious if there is any Clippy rule that can do this check?

Dunqing commented:
### Merge activity

* **Nov 1, 12:36 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(ast): `StaticMemberExpression::get_first_object` use loop instead of recursion (#7065)

Follow-on after #6969. Pure refactor. Loops are generally cheaper than recursion.
